### PR TITLE
i/b/shared-memory: handle "private" plug attribute in shared-memory interface correctly

### PR DESCRIPTION
**Pull request by MiguelPires**
The isPrivate() method assumes that a plug has been sanitised by BeforePreparePlug and panics if there's no "private" attribute. There's been a customer report of this happening and even though it's unclear how the attribute is missing/wrong, we should handle it as gracefully as we can instead of panicking.

### CODE REVIEW

File: interfaces/builtin/shared_memory.go [hunk 1]
```diff
@@ -321,7 +326,10 @@ func (iface *sharedMemoryInterface) AppArmorConnectedSlot(spec *apparmor.Specifi
 }
 
 func (iface *sharedMemoryInterface) MountConnectedPlug(spec *mount.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
-	if !iface.isPrivate(plug) {
+	private, err := iface.isPrivate(plug)
+	if err != nil {
+		return err
+	} else if !private {
```
Comment (olivercalder) [inline interfaces/builtin/shared_memory.go:332]: (very nitpick) I'm personally not a huge fan of using `else if` after an `if` statement that always returns -- I prefer a new separate `if` for the next condition that might require a return, so the error handling case is more semantically separated from the `!private`, which is not an error case. But the latter is more verbose, and this is really just personal preference on my part.

**Review by olivercalder — APPROVED** [review]
This looks great, thanks!

File: interfaces/builtin/shared_memory.go (hunk 1 above)
Comment (mvo5) [reply interfaces/builtin/shared_memory.go:332]: Fwiw @olivercalder, the "effective go" section about "if" https://go.dev/doc/effective_go#if has a similar recommendation/preference

Comment (MiguelPires) [reply interfaces/builtin/shared_memory.go]: That's a good suggestion, thanks! I've split it now

**Review by olivercalder — APPROVED** [review]
LGTM, thanks!

**Comment by mvo5** [comment]
Fwiw, I ran the update spread test against master (without the fix but with panic(err) to see what the panic is) and get the expected crash:
```
# qemu-nested:ubuntu-16.04-64 .../tests/nested/core/core-snap-plug-panic# remote.exec sudo journalctl -u snapd
....
Aug 23 17:14:34 localhost snapd[1393]: panic: snap "shm-plug" does not have attribute "private" for interface "shared-memory"
Aug 23 17:14:35 localhost snapd[1393]: goroutine 1 [running]:
Aug 23 17:14:35 localhost snapd[1393]: github.com/snapcore/snapd/interfaces/builtin.(*sharedMemoryInterface).isPrivate(0x0?, 0x5?)
...
```
and with the fix it works when running the nested test in qemu locally.

**Comment by mvo5** [comment]
I removed the spread test again as it's now part of https://github.com/snapcore/snapd/pull/13111 - while 13111 fixes this also in a different way I would still like to get rid of the panic, maybe @pedronis can have a look (no rush there though as the fix is now ready)

**Review by pedronis — APPROVED** [review]
this looks like an ok change anyway

**Pull request merged**